### PR TITLE
77 add the create game endpoint

### PR DESCRIPTION
**Pull request by yusufzartit**

### CODE REVIEW

File: src/main/java/com/iotiq/application/service/GameService.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        // Create GamerGame for the gamer as CREATOR
+        GamerGame gamerGame = gamerGameService.createGamerGame(gamer, game);
+
+        // Establish bidirectional associations
+        gamerGameService.associateGameWithGamer(gamer, game, gamerGame);
+
+        // Create GamerQuestion for each question
+        List<GamerQuestion> gamerQuestion = gamerQuestionService.createGamerQuestionList(questions);
+
+        // Establish bidirectional associations for questions
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/service/GameService.java:52]: Why are you doing this seperately?

File: src/main/java/com/iotiq/application/service/GameService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        questionRepository.saveAll(questions);
+        
+        // Create a Game entity
+        Game game = createGameEntity(request);
+
+        // Create GamerGame for the gamer as CREATOR
+        GamerGame gamerGame = gamerGameService.createGamerGame(gamer, game);
+
+        // Establish bidirectional associations
+        gamerGameService.associateGameWithGamer(gamer, game, gamerGame);
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/service/GameService.java:47]: Why are you doing this seperately?

File: src/main/java/com/iotiq/application/service/GameService.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        List<GamerQuestion> gamerQuestion = gamerQuestionService.createGamerQuestionList(questions);
+
+        // Establish bidirectional associations for questions
+        gamerQuestionService.associateQuestionsWithGamer(questions, gamerQuestion, gamerGame);
+
+        // Save the game
+        return gameRepository.save(game);
+    }
+
+    private Game createGameEntity(GameRequest request) {
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/service/GameService.java:59]: You may rename this as mapRequestToEntity or sth like it. There are similar names around

File: src/main/java/com/iotiq/application/service/GameService.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    private final GamerQuestionService gamerQuestionService;
+
+    @Transactional
+    public Game createGame(GameRequest request) {
+        // Retrieve the currently logged-in user
+        User user = userService.getCurrentUser();
+
+        // Get or create a Gamer entity associated with the user
+        Gamer gamer = gamerService.getOrCreateGamerForUser(user);
+
```
[the remark this answers is not in the document]
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/service/GameService.java:35]: #82 issue has been opened for this suggestion.

File: src/main/java/com/iotiq/application/service/GameService.java (hunk 1 above)
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/service/GameService.java]: I used them to improve code readability

Comment (yusufzartit) [reply src/main/java/com/iotiq/application/service/GameService.java]: "createGamerQuestionList" and "associateQuestionsWithGamer" with a "createGamerQuestionListAndAssociateWithQuestions" has been replaced.

File: src/main/java/com/iotiq/application/service/GameService.java (hunk 2 above)
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/service/GameService.java]: "createGamerGame" and "associateGameWithGamer" with a "createGamerGameAndAssociateWithGameAndGamer" has been replaced.

File: src/main/java/com/iotiq/application/service/GameService.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        // Retrieve a list of questions based on the request parameters
```
[the remark this answers is not in the document]
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/service/GameService.java]: The method related to "getQuestion" has been removed from here

File: src/main/java/com/iotiq/application/domain/Game.java [hunk 6]
```diff
@@ -22,5 +22,5 @@ public class Game extends AbstractPersistable<UUID> {
     private GameStatus status;
     @OneToMany(cascade = CascadeType.ALL)
     @JoinColumn(name = "game_id")
-    private Set<GamerGame> gamerGame = new HashSet<>();
+    private Set<GameGamer> gameGamer = new HashSet<>();
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/domain/Game.java:25]: better is gameGamers. It's a set.

File: src/main/java/com/iotiq/application/service/GameGamerService.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import com.iotiq.application.domain.Gamer;
+import com.iotiq.application.domain.GamerRole;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class GameGamerService {
+
+    public GameGamer createGameGamerAndAssociateWithGameAndGamer(Gamer gamer, Game game) {
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/service/GameGamerService.java:14]: No need to create A Game gamer and not associate with a game. So I think "createGameGamer" is enough

**Pull request merged**